### DISCUSSION TITLE
Sharing session in domain and sub-domain

### DISCUSSION THREAD

**Discussion by maqsudinamdar** [laravel/framework#36276] [Q&A]
I've two Laravel application on same server but on different path. And I want to merge both application by sharing session. 

Both of these application has domain and sub-domain combination namely:
- `example.com`
- `tool.example.com`

And I'd like authenticate user in `tool.example.com` from authenticating in `example.com` using following [articles.][1]

---
> ## **Exception has occurred.** Illuminate\Auth\AuthenticationException: Unauthenticated.

---

## I tried the following:

 1. ***Sharing the cookies, sessions and users between the applications*** 
 2. ***Use the same `APP_KEY`*** 
 3. ***Use the same domain***
 
---

### `example.com` Laravel application I did the following:

`.env`
```
DB_DATABASE=common_database
SESSION_DRIVER=database
APP_KEY=same_key
SESSION_DOMAIN=".example.com"
```  
- Storing session in database. [link][2]

- In `example.com` we're using `example` **Guards** with `example` **Provider** using `Example` **Model** to authenticate web request.

`config/auth.php`
```
    'guards' => [
        'example' => [
            'driver' => 'session',
            'provider' => 'example',
        ],
    ],

    'providers' => [
        'example' => [
            'driver' => 'eloquent',
            'model' => App\Example::class,
        ],
    ],
```

---

### Similarly, `tool.example.com` :

`.env`
```
DB_DATABASE=common_database
SESSION_DRIVER=database
APP_KEY=same_key
SESSION_DOMAIN=".example.com"
```  
- In `tool.example.com` i've assigned the `$table` variable as `example` in `User` **Model** 

`User.php`
```
class User extends Authenticatable 
{
    protected $table = 'example';
```
`config/auth.php`
```
    'guards' => [
        'example' => [
            'driver' => 'session',
            'provider' => 'users',
        ],
    ],

    'providers' => [
        'users' => [
            'driver' => 'eloquent',
            'model' => App\User::class,
        ],
    ],
```
>Note: I'm expecting to use `example` table to authenticate request in both application.

---

Any help is greatly appreciated. Thanks! in advance.

  [1]: https://medium.com/@zsolt.gyure96/how-to-share-sessions-between-two-laravel-applications-4b9d061fa599
  [2]: https://laravel.com/docs/8.x/session#database

**Comment by Thomasosho** [comment]
When you discover the answer, please share
